### DISCUSSION TITLE
What's the proper way to parse the return type of 'agtype' from the client side?

### DISCUSSION THREAD

**Discussion by rburton** [apache/age#73] [Q&A] [answered]
I see that in the example, I would get the following results based off this query:

`SELECT * from cypher('my_graph_name', $$ MATCH p=(a)-[]-(b) RETURN p $$) as (a agtype)`

**[{"id": 844424930131969, "label": "Part", "properties": {"part_num": "123"}}::vertex, {"id": 1125899906842625, "label": "used_by", "end_id": 844424930131970, "start_id": 844424930131969, "properties": {"quantity": 1}}::edge, {"id": 844424930131970, "label": "Part", "properties": {"part_num": "345"}}::vertex]::path**

**[{"id": 844424930131970, "label": "Part", "properties": {"part_num": "345"}}::vertex, {"id": 1125899906842625, "label": "used_by", "end_id": 844424930131970, "start_id": 844424930131969, "properties": {"quantity": 1}}::edge, {"id": 844424930131969, "label": "Part", "properties": {"part_num": "123"}}::vertex]::path**

Is there a grammar I should be looking for?

**Comment by emotionbug** [comment, marked as answer]
I know that basically, Agtype is json format.
but, strictly not same with JSON.

additionally, have annotation like `::path`, `::edge`, `::vertex`.
However, it appears that there is no explanation in the document.

anyway... you can implement with ANTLR4 like behind codes.

**References**
https://github.com/apache/incubator-age/blob/master/drivers/Agtype.g4
https://github.com/apache/incubator-age/blob/master/drivers/jdbc/lib/src/main/java/org/apache/age/jdbc/base/AgtypeListener.java

**Reply by rburton** [reply]
Thank you - this is great!
